PocketKit/Sources/PocketKit/Collections/CollectionViewModel.swift

```diff
@@ -207,15 +219,46 @@ class CollectionViewModel {
     }
 }
 
+// MARK: Error / Offline
+extension CollectionViewModel {
+    var errorEmptyState: EmptyStateViewModel {
+        return ErrorEmptyState(featureFlags: featureFlags, user: user)
+    }
+
+    var isOffline: Bool {
+        return networkPathMonitor.currentNetworkPath.status == .unsatisfied
+    }
+
+    private func observeNetworkChanges() {
+        networkPathMonitor.updateHandler = { [weak self] path in
+            DispatchQueue.main.async {
+                self?.handleNetworkChange(path.status)
+            }
+        }
+    }
+
+    /// Observes network changes and fetch collection when user returns online
+    private func handleNetworkChange(_ status: NWPath.Status) {
+        guard status != currentNetworkStatus else {
+            return
+        }
+
+        if currentNetworkStatus == .unsatisfied, status == .satisfied {
+            self.snapshot = Self.loadingSnapshot()
+            fetch()
```

Comment:
Hey @cyndichin , since we are storing the list locally, I am not sure that we should re-fetch the collection simply because users could go temporarily offline, which could cause an overhead that, most likely, will provide the same data that we have locally, what do you think?

Reply:
Oh, I thought we were storing the list locally only when their user opens the collection. In the case here, when the user is offline and never fetches the collection and then resumes online, I was thinking we should try to fetch the collection again. I do expect us to do a check for the local store first and then fetch remotely if not available.